packages/scm/src/browser/scm-commit-widget.tsx

```diff
@@ -0,0 +1,170 @@
+/********************************************************************************
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { Message } from '@phosphor/messaging';
+import { SelectionService } from '@theia/core/lib/common';
+import * as React from 'react';
+import TextareaAutosize from 'react-autosize-textarea';
+import { ScmInput } from './scm-input';
+import {
+    ContextMenuRenderer, ReactWidget, LabelProvider, KeybindingRegistry, StatefulWidget} from '@theia/core/lib/browser';
+import { ScmService } from './scm-service';
+
+@injectable()
+export class ScmCommitWidget extends ReactWidget implements StatefulWidget {
+
+    static ID = 'scm-commit-widget';
+
+    @inject(ScmService) protected readonly scmService: ScmService;
+    @inject(SelectionService) protected readonly selectionService: SelectionService;
+    @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
+    @inject(KeybindingRegistry) protected readonly keybindings: KeybindingRegistry;
+
+    protected shouldScrollToRow = true;
+
+    /** don't modify DOM use React! only exposed for `focusInput` */
+    protected readonly inputRef = React.createRef<HTMLTextAreaElement>();
+
+    constructor(
+        @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer,
+    ) {
+        super();
+        this.scrollOptions = {
+            suppressScrollX: true,
+            minScrollbarLength: 35
+        };
+        this.addClass('theia-ScmCommit');
+        this.id = ScmCommitWidget.ID;
+    }
+
+    @postConstruct()
+    protected init(): void {
+
+    }
+
+    protected onActivateRequest(msg: Message): void {
+        super.onActivateRequest(msg);
+        this.focus();
+    }
+
+    public focus(): boolean {
+        (this.inputRef.current || this.node).focus();
+        return true;   // returns false to set parent node focus - tidy this up
```

Comment:
could not find any clients expecting boolean

Reply:
Yes, I guess that is what I thought when I added the 'tidy this up' comment.  It now returns void.